test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/ViewFeatures/ViewDataDictionaryOfTModelTest.cs

```diff
@@ -124,33 +124,50 @@ public void CopyConstructors_InitalizeModelAndModelMetadataBasedOnSource_ModelOf
-            Assert.Same(source.ModelMetadata, viewData1.ModelMetadata);
-            Assert.Equal(source.Count, viewData1.Count);
-            Assert.Equal("bar", viewData1["foo"]);
-            Assert.IsType<CopyOnWriteDictionary<string, object>>(viewData1.Data);
+            Assert.NotNull(viewData.ModelState);
+            Assert.NotNull(viewData.TemplateInfo);
+            Assert.Equal("prefix", viewData.TemplateInfo.HtmlFieldPrefix);
+            Assert.NotSame(source.TemplateInfo, viewData.TemplateInfo);
+            Assert.Same(model, viewData.Model);
+            Assert.NotNull(viewData.ModelMetadata);
+            Assert.Equal(typeof(object), viewData.ModelMetadata.ModelType);
+            Assert.Same(source.ModelMetadata, viewData.ModelMetadata);
+            Assert.Equal(source.Count, viewData.Count);
+            Assert.Equal("bar", viewData["foo"]);
+            Assert.IsType<CopyOnWriteDictionary<string, object>>(viewData.Data);
+        }
 
-            Assert.NotNull(viewData2.ModelState);
-            Assert.NotNull(viewData2.TemplateInfo);
-            Assert.Equal("prefix", viewData2.TemplateInfo.HtmlFieldPrefix);
-            Assert.NotSame(source.TemplateInfo, viewData2.TemplateInfo);
-            Assert.Same(model, viewData2.Model);
-            Assert.NotNull(viewData2.ModelMetadata);
-            Assert.Equal(typeof(SupremeTestModel), viewData2.ModelMetadata.ModelType);
-            Assert.Same(source.ModelMetadata, viewData2.ModelMetadata);
-            Assert.Equal(source.Count, viewData2.Count);
-            Assert.Equal("bar", viewData2["foo"]);
-            Assert.IsType<CopyOnWriteDictionary<string, object>>(viewData2.Data);
+        [Fact]
+        public void CopyConstructor_InitializesModelBasedOnSource_ModelMetadataBasedOnTModel()
+        {
+            // Arrange
+            var metadataProvider = new EmptyModelMetadataProvider();
+            var model = new SupremeTestModel();
+            var source = new ViewDataDictionary(metadataProvider)
+            {
+                Model = model,
+            };
+            source["foo"] = "bar";
+            source.TemplateInfo.HtmlFieldPrefix = "prefix";
+
+            // Act
+            var viewData = new ViewDataDictionary<TestModel>(source);
+
+            // Assert
+            Assert.NotNull(viewData.ModelState);
+            Assert.NotNull(viewData.TemplateInfo);
+            Assert.Equal("prefix", viewData.TemplateInfo.HtmlFieldPrefix);
+            Assert.NotSame(source.TemplateInfo, viewData.TemplateInfo);
+            Assert.Same(model, viewData.Model);
+            Assert.NotNull(viewData.ModelMetadata);
+            Assert.Equal(typeof(TestModel), viewData.ModelMetadata.ModelType);
+            Assert.NotSame(source.ModelMetadata, viewData.ModelMetadata);
+            Assert.Equal(source.Count, viewData.Count);
+            Assert.Equal("bar", viewData["foo"]);
+            Assert.IsType<CopyOnWriteDictionary<string, object>>(viewData.Data);
```

Comment:
Would be way clearer if these tests were focused on what's important about each distinct test. As it is, this is an insurmountable mountain of `Assert`s.